src/gpio/wait_asynch.rs

```diff
@@ -0,0 +1,110 @@
+use core::{
+    future::Future,
+    sync::atomic::{AtomicBool, Ordering},
+    task::Poll,
+};
+extern crate alloc;
+use crate::gpio::{InputMode, InterruptType, Pin, PinDriver};
+use alloc::sync::Arc;
+use atomic_waker::AtomicWaker;
+use esp_idf_sys::{esp_nofail, gpio_intr_disable, EspError};
+
+impl<T: Pin, MODE: InputMode> embedded_hal_async::digital::Wait for PinDriver<'_, T, MODE> {
```

Comment:
Please add the async methods also directly on the `PinDriver` struct and delegate to those from here. The recent philosophy of `esp-idf-hal` is that everything doable via the e-hal traits (blocking or sync) should also be doable via an `esp-idf-hal`-specific API on the `*Driver` structs as well. This has the extra benefit that folks not willing to use nightly features can opt out of those, at the price of depending on `esp-idf-hal`-specific API.

Reply:
Will do!